app/views/lookup_keys/_value.html.erb

```diff
@@ -9,7 +9,7 @@
     <%= text_field_tag '', nil, :class => 'matcher_value' %>
   </td>
   <td>
-    <%= hidden_value_field(f, :value, f.object.value, f.object.use_puppet_default, :'data-property' => 'value') %>
+    <%= hidden_value_field(f, :value, f.object.value_before_type_cast, f.object.use_puppet_default, :'data-property' => 'value') %>
```

Comment:
this isn't a regression.
if the value is invalid (but key's default value is valid), it acts like it should.

if key is invalid, the value doesn't even get its validations run (thanks DHH)
this seems like a AR/AM bug, and it's present in develop. we should fix it, but let's do it in a different task.